
\section{Introduction}
Profiling gene expression is common problem in modern biology. It is also a classic $n \ll p$ example, inspiring statistical advances in multiple testing. Because the presence of a meaningful biological effect fits awkwardly into the NHST framework, arguably most questions of interest are better addressed through estimation \citet{deseq2014}. This is the approach that we pursue here.

Current estimation methodologies can be understood as improving upon a ``straight" estimator by modeling gene-specific model parameters to borrow information across genes. \textit{Insert analogy to Stein's estimator}.

Because there are many genes to observe, a gene expression experiment presents an opportunity to learn the underlying distribution of the gene-specific parameters. Herein lies dragons... In fact, given the large number of genes, G, the parameters of this distribution, if it should be assumed to have a simple form, will tend to be estimated very precisely. If the model is overly simplistic, the information borrowed across genes will tend to be quite vague. The impact of the hierarchical modeling approach is to ``regularize" inference for gene-specific parameters by shrinking the posterior distribution of these parameters away from the tails of the hierarchical distribution. Because of its essential role, the shape of the tail, vis a vis the choice of the hierarchical model, should be considered.

In \citet{voom}, the authors proposed using a nonparametric estimate of the mean variance relationship using the straight estimates as data to produce precision estimates for each datum, thereby borrowing information about the measure of uncertainty within a gene. In Niemi et al, 2016, the authors used an empirical Bayes approach that considered the marginal empirical distributions of the straight estimators to inform the selection of prior distributions of the regression coefficients.

\citet{liu}

\section{Gene expression data}
\subsection{Normalization and weighting}
\paragraph{voom}
Theory suggests that an RNA-seq count, $C$, would have a Poisson distribution, with variance equal to the mean ($\op{V}(C)=\op{E{C}$), if the RNA from the same sample were sequenced repeatedly. Biological variation (either between subjects or through repeated sampling) leads overdispersion of the counts. Often, the negative binomial distribution is used because its variance is quadratic with the mean ($\op{V}(C)=\op{E}+\phi\op{E}^2(C)$). Because gene expression is usually measured on the log scale, through an application of the delta method, we might re-express this in terms of the log-count ($\op{V}(\log C) \approx \frac{1}{\op{E}(C)} + \phi$). While this assumption may be adequate for genes with similar expression levels, empirical studies suggest that the overdispersion parameter, $\phi$, tends to be larger for genes with lower expression than those with higher expression.

\cite{voom} proposed \textit{voom}, a way to incorporate between-gene information about the mean-variance relationship in an RNA-seq data set through precision weighting of individual log-counts, making the data amenable to analysis via a linear model. It works by first calculating, for each gene, a sample standard deviation and mean log-count, and then fitting a LOWESS curve, treating the square root standard deviations as a response and the mean log-count as the predictor. Fitted values for all counts are converted into precision weights.


\section{Bayesian nonparametric model}

\section{Simulation study 1}

\paragraph{Construction of simulated data:}
\begin{enumerate}
\item Select random draw $\mathcal{P}^{(s)}$ from posterior distribution of $\mathcal{P}$

\item Sample $(\beta_g,\sigma^2_g)$ from $\mathcal{P}^{(s)}$

\item Sample $y_{g,rep} \sim N(X\beta_g,\sigma^2_g)$
\end{enumerate}
Produce 6-10 such data sets.\\

\paragraph{Analysis}

Run Gibbs sampler for each data set:
\begin{enumerate}
\item look at coverage of posterior credible intervals for gene-specific parameters

\item look at calibration of posterior probabilities for proposition(s) of interest

\item compare MSE to unpooled/unbiased estimateors
\end{enumerate}
\section{Simulation study 2}
Generally: Generate count data, truth estimated from model, and compare voom pipelines.

Outputs: ROC curves, difference histograms

\section{Analysis of Paschold data}




