%&latex
\documentclass[12pt]{article}
\usepackage{amsmath,amsfonts,bm}
\usepackage{graphicx,psfrag,epsf}
\usepackage{enumerate}
\usepackage{natbib}
\usepackage{url} % not crucial - just used below for the URL 
\usepackage{etoolbox}
\usepackage{booktabs}
\usepackage{multirow}
\usepackage{lscape}
%\pdfminorversion=4
% NOTE: To produce blinded version, replace "0" with "1" below.
\newcommand{\blind}{0}

% DON'T change margins - should be 1 inch all around.
\addtolength{\oddsidemargin}{-.5in}%
\addtolength{\evensidemargin}{-.5in}%
\addtolength{\textwidth}{1in}%
\addtolength{\textheight}{1.3in}%1.3?
\addtolength{\topmargin}{-.8in}%
\graphicspath{{figures_tables/}}

\newcommand{\op}{\operatorname}
\newcommand{\ind}{\stackrel{ind.}{\sim}}

\begin{document}
\newtoggle{thesis}
\toggletrue{thesis}
% \bibliographystyle{natbib}

\def\spacingset#1{\renewcommand{\baselinestretch}%
{#1}\small\normalsize} \spacingset{1}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\if1\blind
{
  \title{\bf A Bayesian nonparametric analysis of RNA-seq gene expression data}
  \author{Author 1\thanks{
    The author gratefully acknowledges Dr. Jarad Niemi for advise and comments.\textit{This project was funded in part by a grant from the NIH, XXXX.}}\hspace{.2cm}\\
    Department of Statistics, Iowa State University}
  \maketitle
} \fi

\if0\blind
{
  \bigskip
  \bigskip
  \bigskip
  \begin{center}
    {\LARGE\bf A Bayesian nonparametric analysis of RNA-seq gene expression data}
\end{center}
  \medskip
} \fi

\bigskip
\begin{abstract}
We present a Bayesian nonparametric model suitable for gene expression profiling using high-throughput sequencing technologies. An analysis pipeline is detailed for the analysis of RNA-seq data and we apply it to an experiment done on several maize genotypes. This case study highlights an advantage that Bayesian methods enjoy --- that evaluating the probability of arbitrary hypotheses regarding the model parameters is straightforward. Simulations studies are conducted which suggest that our method provides superior parameter estimates and is able to improve the accuracy of gene classification compared to other, popular methods. To make our method computationally feasible we implemented a blocked Gibbs sampler exploiting fine grain parallelism on general purpose graphical processing units.

\end{abstract}

\noindent%
{\it Keywords:}  Dirichlet process, hierarchical modeling, parallel computing
\vfill

\newpage
\tableofcontents

\spacingset{1.45} % DON'T change the spacing!

\include{chapter1}

\appendix
\include{appendix}
\bigskip
\begin{center}
{\large\bf SUPPLEMENTARY MATERIAL}
\end{center}

\begin{description}

\item[Title:] Brief description. (file type)

\item[R-package for  MYNEW routine:] R-package ?MYNEW? containing code to perform the diagnostic methods described in the article. The package also contains all datasets used as examples in the article. (GNU zipped tar file)

\item[HIV data set:] Data set used in the illustration of MYNEW method in Section~ 3.2. (.txt file)

\end{description}


\bibliographystyle{plainnat}

\bibliography{methods}
\end{document}
